drizzlepac/haputils/catalog_utils.py

```diff
@@ -1486,6 +1621,125 @@ def _define_total_table(self, updated_table):
+        may be of poor quality.
+
+        If so, try using the RickerWavelet2DKernel for source detection.
+
+        Parameters
+        ----------
+        segm_img : Segmentation image
+            Segmentation image created by the Photutils package by detect_sources ()
+
+        detection_image :  FITS data
+            The total drizzled detection image (aka white light data).
+
+        big_island_only : bool, optional
+            Test for 'big island' situations only? (True/False)
+            Default value is False.
+
+        max_biggest_source : float, optional
+            Maximum limit on the single largest detected "source".
+            Default value is 0.015.
+
+        max_source_fraction : float, optional
+            Maximum limit on the fraction of pixels identified as part of a "source".
+            Default value is 0.075.
+
+
+        Returns
+        -------
+        is_poor_quality: bool
+            Determination as to whether or not the science field is crowded or contains big islands.
+
+        """
+
+        is_poor_quality = False
+        is_poor_quality = self._seg_rejection(segm_img,
+                                              detection_image,
+                                              big_island_only=big_island_only,
+                                              max_biggest_source=max_biggest_source,
+                                              max_source_fraction=max_source_fraction)
+
+        return is_poor_quality
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    def _seg_rejection(self, segm_img, image_data, big_island_only=False, max_biggest_source=0.015, max_source_fraction=0.075):
+        """
+        Determine if the largest "source" or if the total fraction of "sources" exceeds a threshold.
+
+        Identify situations where the largest source exceeds a user-specified fraction of all image
+        pixels and / or situations where the total source fraction exceeds a user-specified fraction
+        of the image (aka 'big islands')
+
+        Algorithm is essentially the standalone function written by R.L.White (STScI) for the 
+        Hubble Legacy Archive (HLA).
+
+        Parameters
+        ----------
+        segm_img : Segmentation image
+            Segmentation image created by the Photutils package by detect_sources ().
+
+        image_data :  FITS data
+            The total drizzled detection image (aka white light data).
+
+        big_island_only : bool, optional
+            Test for 'big island' situations only? (True/False)
+            Default value is False.
+
+        max_biggest_source : float, optional
+            Maximum limit on the single largest detected "source".
+            Default value is 0.015.
+
+        max_source_fraction : float, optional
+            Maximum limit on the fraction of pixels identified as part of a "source".
+            Default value is 0.075.
```

Comment:
Not necessary to state this since the values are shows in the function/method signature above. Can delete this line.